aten/src/ATen/native/native_functions.yaml

```diff
@@ -5467,6 +5467,13 @@
     CPU: foreach_add_scalar_kernel_fallback
     CUDA: foreach_tensor_add_scalar_kernel_cuda
 
+- func: _foreach_add_.Scalar(Tensor[](a!) self, Scalar scalar) -> Tensor[](a!)
+  device_guard: False  
+  variants: function
+  dispatch:
+    CPU: foreach_add_scalar_kernel_fallback_
+    CUDA: foreach_tensor_add_scalar_kernel_cuda_
```

Comment:
these having different names is really confusing, it sounds like the CPU variant doesn't involve tensors.

Reply:
Also using fallback is confusing because a fallback kernel is a different concept entirely.  I'd suggest using "slow"

Reply:
Renamed
```
CPU: foreach_tensor_add_scalar_kernel_slow_
CUDA: foreach_tensor_add_scalar_kernel_cuda_
```